### DISCUSSION TITLE
Issue with jQuery peer-deps

### DISCUSSION THREAD

**Discussion by RBilly** [web-infra-dev/rsbuild#4094] [Q&A]
Hello
I'm in progress of migrating an "old" monorepo based on Craco v7 to rsbuild.
I was able to resolve most issues myself but there is one that I'm not able to solve without refactoring the code of almost all modules of the monorepo.
As the title suggest, I have a common component that relies on jQuery: 
https://github.com/skratchdot/react-bootstrap-daterangepicker/tree/4.1.0
Best solution would obviously be to use something else, but reality is that it's used in a lot of place and I'd rather make rsbuild work with it right now than wait for all modules to move away from it.

Does someone have any idea how to make rsbuild play nice with it ?

here is my error coming from this component: this.$picker.daterangepicker is not a function

What I've tried:
- updating react-bootstrap-daterangepicker dependency gradually up to latest version
- deduping using pnpm
- using this in rsbuild config file:
```js
output: {
      externals: {
        $: 'jquery',
        jQuery: 'jquery'
      },
    },
```
I also saw this from the rspack doc: https://rspack.dev/plugins/webpack/provide-plugin#jquery
But I don't know how to use `new rspack.ProvidePlugin` in rsbuild config nor if it would help

I will be thankful for any help, I would be very sad to be forced to hold the migration because of jquery dependency 😞

**Comment by 9aoy** [comment]
you can get rspack instance and set rspack configuration in [tools.rspack](https://rsbuild.dev/config/tools/rspack#rspack).

**Reply by RBilly** [reply]
Thanks, I misread the doc.
I added this to my rsbuild config
```js
tools: {
      rspack: (config, {rspack}) => {
        config.plugins?.push(new rspack.ProvidePlugin({
          $: 'jquery',
          jQuery: 'jquery',
        }));
        return config;
      }
    }
````
But I still get the same error.
Unfortunately I don't have any more experience with Webpack than Rsbuild/Rspack, so I don't understand why it was working with Craco and does not with Rsbuild.
